### PR TITLE
Add react-typescript import & export, use effect

### PR DESCRIPTION
**Pull request by mildronize**
Please review my PR. 😁

### CODE REVIEW

File: content/frontend-web/react-typescript/design-pattern.md [hunk 1]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+- Re-export, then group all imports and rename
+
+    ```ts
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // MyParentComponent.tsx
+    export  * as MyComponent from "./MyComponent";
+    // App.tsx
+    import { MyComponent } from './MyComponent';
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Should it be `from './MyParentComponent`?

File: content/frontend-web/react-typescript/design-pattern.md [hunk 2]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+        </>
+    );
+    ```
+
+
+- Export type
+
+    ```ts
+    // MyComponent.tsx
+    export type Sex = "Male" | "Female"; // Accept only "Male" or "Female"
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Just a minor change, it think when taking about male or female we prefer to use `gender`.
https://www.oxfordlearnersdictionaries.com/definition/english/gender?q=gender

File: content/frontend-web/react-typescript/design-pattern.md [hunk 3]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+- Re-export all
+
+    ```ts
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // MyParentComponent.tsx
+    export * from "./MyComponent";
+    // App.tsx
+    import { Switch, Group } from './MyComponent';
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Should it be `import { Switch, Group } from './ParentComponent';`?

File: content/frontend-web/react-typescript/design-pattern.md [hunk 4]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+    );
+    ```
+
+- Re-export all
+
+    ```ts
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // MyParentComponent.tsx
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md:67]: Could you please add one empty line to separate each component and apply this to everywhere in these examples?

File: content/frontend-web/react-typescript/design-pattern.md [hunk 5]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+    // MyComponent.tsx
+    export interface User {
+        id: string;
+        name: string;
+    }
+    // App.tsx
+    import { User } from './MyComponent'
+    ```
+
+
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Very minor, I think only one empty line before a new header should be okay.

File: content/frontend-web/react-typescript/design-pattern.md [hunk 6]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+    export type Sex = "Male" | "Female"; // Accept only "Male" or "Female" 
+    // App.tsx
+    import { Sex } from './MyComponent'
+    ```
+
+- Export interface
+
+    ```ts
+    // MyComponent.tsx
+    export interface User {
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Man, here is one thing that I don't like TS naming convention which is different from C#.
In C#, we use `I` prefix for a interface's name, e.g. `IUser`, `ICollection`. 
This help us to distinguish which one is a class or an interface. 

Just FYI.

File: content/frontend-web/react-typescript/design-pattern.md [hunk 7]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+- ทำงานเมื่อ dependencies มีการเปลี่ยนแปลง (Running when specific dependencies change)a
+
+## ทำงานเฉพาะครั้งแรกที่ Component mount
+
+จะทำครั้งแรก ครั้งเดียวเมื่อ Component โหลดเสร็จ
+
+```ts
+function FirstExecuteComponent() {
+
+  // add event listeners (Flux Store, WebSocket, document, etc.)
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md]: Very minor, please prefer to start a comment with A capital alphabet like a normal sentence.

File: content/frontend-web/react-typescript/design-pattern.md [hunk 8]
```diff
@@ -6,6 +6,217 @@ showToc: true
...
+  useEffect(
+    () => console.log("Component loaded"),
+    []
+  );
+  return null;
+}
+```
+
+ข้อสังเกตุ `useEffect` ต้องการรับ 2 parameters ถ้าอยากให้เหมือน `componentDidMount()` ให้ใส่ Array เปล่าๆ ใน parameters ตัวที่สอง `[]`
```
Comment (aaronamm) [inline content/frontend-web/react-typescript/design-pattern.md:146]: 👍 Good one.

**Review by aaronamm — CHANGES_REQUESTED** [review]
Your PR is good.
After you have fixed my feedbacks, I will merge it.

BTW, I have some suggestions and you don't need to do it now.

1) If you can add `how to deal with a function component parameter`, it would be great.
For example.
- strongly type `props`.
- destruction `props`
- default value `props`
- alias/change props name inside a function
2) React function component common mistakes and how to fix them.

File: content/frontend-web/react-typescript/design-pattern.md (hunk 1 above)
Comment (mildronize) [reply content/frontend-web/react-typescript/design-pattern.md]: Yes, it's wrong.

File: content/frontend-web/react-typescript/design-pattern.md (hunk 2 above)
Comment (mildronize) [reply content/frontend-web/react-typescript/design-pattern.md]: Good word.

File: content/frontend-web/react-typescript/design-pattern.md (hunk 4 above)
Comment (mildronize) [reply content/frontend-web/react-typescript/design-pattern.md:67]: Sure.

File: content/frontend-web/react-typescript/design-pattern.md (hunk 5 above)
Comment (mildronize) [reply content/frontend-web/react-typescript/design-pattern.md]: OK

File: content/frontend-web/react-typescript/design-pattern.md (hunk 6 above)
Comment (mildronize) [reply content/frontend-web/react-typescript/design-pattern.md]: Yes, I should start with `I`.

**Comment by mildronize** [comment]
> BTW, I have some suggestions and you don't need to do it now.
> 1. If you can add how to deal with a function component parameter, it would be great.
For example.
>     - strongly type props.
>     - destruction props
>     - default value props
>     - alias/change props name inside a function
> 2. React function component common mistakes and how to fix them.

@aaronamm clear review & good suggestion. thanks a lot.

**Review by aaronamm — APPROVED** [review]

**Pull request merged**